### DISCUSSION TITLE
UT-Assert Framework for custom unit testing implementation

### DISCUSSION THREAD

**Discussion by quinnha** [nasa/cFS#362] [Q&A] [answered]
Hello all,

I was just wondering if anyone had any experience using the UT-Assert library for other non-cFS apps? Thinking specifically for satellite applications, but if anyone had any experience/documentation they'd be willing to share, that would be great!

- Quinn

**Comment by skliper** [comment, marked as answer]
Are these applications using OSAL?  We do use https://github.com/nasa/osal/tree/main/ut_assert as the basis for all the test applications within OSAL (lower level than cFS apps), both functionally here https://github.com/nasa/osal/tree/main/src/tests and in a coverage test context here https://github.com/nasa/osal/tree/main/src/unit-test-coverage.  The pattern should be applicable to anything that uses OSAL.

I'm not aware of any use of ut_assert referenced above independent of OSAL, it uses OS_BSP_* calls to set things (console, etc) up so you'd need to implement those and provide your own BSP to start things up correctly and call into the OS_Application_Startup and Run functions.
